src/MorseL/Internal/HubMethodDiscoverer.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+
+namespace MorseL.Internal
+{
+    internal class HubMethodDiscoverer<THub, TClient> where THub : Hub<TClient>
```

Comment:
I like this abstraction - could potentially derive an interface from its methods for mocking and testing?

Reply:
I created #53 for this.